classes/ValidateOrder.php

```diff
@@ -67,22 +67,40 @@ public function __construct($paypalOrderId, $merchantId)
      */
     public function validateOrder($payload)
     {
+        // API call here
         $paypalOrder = new PaypalOrder($this->paypalOrderId);
         $order = $paypalOrder->getOrder();
+
+        if (empty($order)) {
+            // @todo quickfix : Call API return nothing or fail
+            $message = sprintf('Unable to retrieve Paypal Order for %s', $this->paypalOrderId);
+            \PrestaShopLogger::addLog($message, 1, null, null, null, true);
+            throw new \PrestaShopException($message);
+        }
+
         $apiOrder = new Order(\Context::getContext()->link);
 
         switch ($paypalOrder->getOrderIntent()) {
             case self::INTENT_CAPTURE:
+                // API call here
                 $response = $apiOrder->capture($order['id'], $this->merchantId);
                 break;
             case self::INTENT_AUTHORIZE:
+                // API call here
                 $response = $apiOrder->authorize($order['id'], $this->merchantId);
                 break;
             default:
-                throw new \Exception(sprintf('Unknown Intent type %s', $paypalOrder->getOrderIntent()));
+                // @todo quickfix
+                $message = sprintf('Unknown Intent type %s, Paypal Order %s', $paypalOrder->getOrderIntent(), $this->paypalOrderId);
+                \PrestaShopLogger::addLog($message, 1, null, null, null, true);
+                throw new \PrestaShopException($message);
```

Comment:
What is the reason for using `PrestaShopException` instead of `Exception`?

In a later version, we could implement our own `PsCheckoutException` who would handle the logging by itself.

Reply:
Due to this https://github.com/PrestaShopCorp/ps_checkout/blob/master/controllers/front/DispatchWebHook.php#L113

Reply:
All right!